src/score.h

```diff
@@ -261,16 +305,71 @@ struct ScoreFunction
                     logToConsole(log);
                     return false;
                 }
+
+                if (!allocatePool(maxDuration * sizeof(PoolSynapseData), (void**)&(cb._poolSynapseTickData)))
+                {
+                    CHAR16 log[256];
+                    setText(log, L"Failed to allocate memory for pool synapse data! Try to allocated ");
+                    appendNumber(log, maxDuration * sizeof(PoolSynapseData) / 1024, true);
+                    appendText(log, L" KB");
+                    logToConsole(log);
+                    return false;
+                }
+
+                if (!allocatePool(numberOfOptimizationSteps * sizeof(long long), (void**)&(cb._skipTicks)))
+                {
+                    CHAR16 log[256];
+                    setText(log, L"Failed to allocate memory for skip ticks buffer! Try to allocated ");
+                    appendNumber(log, numberOfOptimizationSteps * sizeof(long long) / 1024, true);
+                    appendText(log, L" KB");
+                    logToConsole(log);
+                    return false;
+                }
+
+                if (!allocatePool(maxDuration * sizeof(long long), (void**)&(cb._ticksNumbers)))
+                {
+                    CHAR16 log[256];
+                    setText(log, L"Failed to allocate memory for ticks number buffer! Try to allocated ");
+                    appendNumber(log, maxDuration * sizeof(long long) / 1024, true);
+                    appendText(log, L" KB");
+                    logToConsole(log);
+                    return false;
+                }
+
+                if (!allocatePool(maxDuration, (void**)&(cb._skipTicksMap)))
+                {
+                    CHAR16 log[256];
+                    setText(log, L"Failed to allocate memory for ticks map buffer! Try to allocated ");
+                    appendNumber(log, maxDuration / 1024, true);
+                    appendText(log, L" KB");
+                    logToConsole(log);
+                    return false;
+                }
             }
         }
 
         for (int i = 0; i < solutionBufferCount; i++) {
             setMem(_computeBuffer[i]._synapses.signs, sizeof(_computeBuffer[i]._synapses.signs[0]) * synapseSignsCount, 0);
             setMem(_computeBuffer[i]._poolSynapseData, sizeof(_computeBuffer[i]._poolSynapseData[0]) * RANDOM2_POOL_SIZE, 0);
             setMem(_computeBuffer[i]._neurons.input, sizeof(_computeBuffer[i]._neurons.input[0]) * allNeuronsCount, 0);
+            setMem(_computeBuffer[i]._ticksNumbers, sizeof(_computeBuffer[i]._ticksNumbers[0]) * maxDuration, 0);
+            setMem(_computeBuffer[i]._skipTicks, sizeof(_computeBuffer[i]._skipTicks[0]) * numberOfOptimizationSteps, 0);
+            setMem(_computeBuffer[i]._skipTicksMap, sizeof(_computeBuffer[i]._skipTicksMap[0]) * numberOfOptimizationSteps, 0);
```

Comment:
Please double-check that you init all the buffers that need init. `_poolSynapseTickData`?
Please also check the size for setMem() of _skipTicksMap. I think it is inconsistent between alloc and setMem().